lib/active_resource/detailed_log_subscriber.rb

```diff
@@ -36,6 +37,22 @@ def warn_on_deprecated_header(event)
         end
       end
     end
+
+    def warn_on_version_eol_warning_header(event)
+      payload = event.payload
+
+      payload[:response].each do |header_name, header_value|
+        case header_name.downcase
+        when 'x-shopify-api-version-warning'
```

Comment:
We are now doing a fair amount of duplicated logic between `warn_on_version_eol_warning_header` and `warn_on_deprecated_header`. Can you see a way to iterate the response headers only once, and ensure that we detect both, one or neither response header if present?